test_regress/vpi_main.cpp

```diff
@@ -0,0 +1,180 @@
+            fprintf(stderr,
+                    "usage: %s [--trace]\n"
+                    "\n"
+                    "Cocotb + Verilator sim\n"
+                    "\n"
+                    "options:\n"
+                    "  --trace      Enables tracing (VCD or FST)\n",
+                    basename(argv[0]));
+            return 0;
+        }
+    }
+
+    Verilated::commandArgs(argc, argv);
+#ifdef VERILATOR_SIM_DEBUG
+    Verilated::debug(99);
+#endif
+    std::unique_ptr<Vtop> top(new Vtop(""));
+    Verilated::fatalOnVpiError(false);  // otherwise it will fail on systemtf
+
+#ifdef VERILATOR_SIM_DEBUG
+    Verilated::internalsDump();
+#endif
+    for (auto it = &vlog_startup_routines[0]; *it != nullptr; it++) {
+        auto routine = *it;
+        routine();
+    }
+    
+    VerilatedVpi::callCbs(cbStartOfSimulation);
+
+#if VM_TRACE
+#if VM_TRACE_FST
+    std::unique_ptr<VerilatedFstC> tfp(new VerilatedFstC);
+    const char* traceFile = "dump.fst";
+#else
+    std::unique_ptr<VerilatedVcdC> tfp(new VerilatedVcdC);
+    const char* traceFile = "dump.vcd";
+#endif
+
+    if (traceOn) {
+        Verilated::traceEverOn(true);
+        top->trace(tfp.get(), 99);
+        tfp->open(traceFile);
+    }
+#endif
+
+    while (!Verilated::gotFinish()) {
+        // Call registered timed callbacks (e.g. clock timer)
+        // These are called at the beginning of the time step
+        // before the iterative regions (IEEE 1800-2012 4.4.1)
+        VerilatedVpi::callTimedCbs();
+
+        // Call Value Change callbacks triggered by Timer callbacks
+        // These can modify signal values
+        settle_value_callbacks();
+
+        // We must evaluate whole design until we process all 'events'
+        bool again = true;
+        while (again) {
+            // Evaluate design
+            top->eval_step();
+
+            // Call Value Change callbacks triggered by eval()
+            // These can modify signal values
+            again = settle_value_callbacks();
+
+            // Call registered ReadWrite callbacks
+            again |= VerilatedVpi::callCbs(cbReadWriteSynch);
+
+            // Call Value Change callbacks triggered by ReadWrite callbacks
+            // These can modify signal values
+            again |= settle_value_callbacks();
+        }
+        top->eval_end_step();
+
+        // Call ReadOnly callbacks
+        VerilatedVpi::callCbs(cbReadOnlySynch);
+
+#if VM_TRACE
+        if (traceOn) {
+            tfp->dump(main_time);
+        }
+#endif
+        // cocotb controls the clock inputs using cbAfterDelay so
+        // skip ahead to the next registered callback
+        const vluint64_t NO_TOP_EVENTS_PENDING = static_cast<vluint64_t>(~0ULL);
+        vluint64_t next_time_cocotb = VerilatedVpi::cbNextDeadline();
+        vluint64_t next_time_timing =
+            top->eventsPending() ? top->nextTimeSlot() : NO_TOP_EVENTS_PENDING;
+        vluint64_t next_time = std::min(next_time_cocotb, next_time_timing);
```

Comment:
```suggestion
        const uint64_t next_time_cocotb = VerilatedVpi::cbNextDeadline();
        const uint64_t next_time_timing =
            top->eventsPending() ? top->nextTimeSlot() : NO_TOP_EVENTS_PENDING;
        const uint64_t next_time = std::min(next_time_cocotb, next_time_timing);
```

No need for vluint*_t as special in C++11 and newer.  Please compile this with -DVL_NO_LEGACY.